src/logger.sh

```diff
@@ -0,0 +1,234 @@
+#!/usr/bin/env bash
+# Copyright (c) YugaByte, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied.  See the License for the specific language governing permissions and limitations
+# under the License.
+#
+
+[[ "${_YB_LOGGER_INCLUDED:=""}" == "yes" ]] && return 0
+_YB_LOGGER_INCLUDED=yes
+
+# -------------------------------------------------------------------------------------------------
+# Global variables used in this module
+# -------------------------------------------------------------------------------------------------
+yb_fatal_quiet=${yb_fatal_quiet:-false}
+yb_log_quiet=${yb_log_quiet:-false}
+FAIL_ON_WARNING=${FAIL_ON_WARNING:-false}
```

Comment:
This should probably be YB_FAIL_ON_WARNING.

Also, in what cases are we using lowercase vs. uppercase names for global variables?

Reply:
All three of these are carry-overs from `yugabyte-bash-common.sh`
```
$ grep -nE 'FAIL_ON_WARNING|yb_log_quiet|yb_fatal_quiet' src/yugabyte-bash-common.sh
34:FAIL_ON_WARNING=${FAIL_ON_WARNING:-false}
282:  if [[ -n "${yb_fatal_quiet:-}" ]]; then
283:    yb_log_quiet=$yb_fatal_quiet
285:    yb_log_quiet=false
289:  if ! "$yb_log_quiet"; then
296:  if [[ ${yb_log_quiet:-} == "true" ]]; then
303:  if [[ ${yb_log_quiet:-} == "true" ]]; then
312:  if [[ ${yb_log_quiet:-} == "true" ]]; then
323:  if [[ ${yb_log_quiet:-} == "true" ]]; then
370:  if ${FAIL_ON_WARNING}; then
460:    yb_log_quiet=false
```